### PR TITLE
Make vicuna finetuning serving self-contained

### PR DESCRIPTION
**Pull request by Michaelvll**
Tested (run the relevant ones):

- [x] Any manual or new tests for this PR (please specify below)
  - [x] `sky launch -c test-vicuna finetune.yaml`
  - [x] `sky launch -c serve serve.yaml`

### CODE REVIEW

File: examples/llm-vicuna/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+[Optional] Finetune a larger model 13B model
+```
+# Train a 13B model instead of the default 7B
+sky spot launch -n vicuna-7b scripts/train-vicuna.yaml --env WANDB_API_KEY --env MODEL_SIZE=13
+
+# Use *unmanaged* spot instances (i.e., preemptions won't get auto-recovered).
+# Unmanaged spot saves the cost of a small controller VM.  We recommend using managed spot as above.
+sky launch -n vicuna scripts/train-vicuna.yaml --env WANDB_API_KEY
```
Comment (concretevitamin) [inline examples/llm-vicuna/README.md:66]: should this be `finetune.yaml`? same for other commands

maybe change to `WANDB_API_KEY=offline` too?

**Review by concretevitamin — COMMENTED** [review]
Awesome @Michaelvll! I tried out the serving part and it worked well. Launching the finetuning part now.

File: examples/llm-vicuna/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+sky launch -c vicuna-serve -s serve.yaml --env MODEL_SIZE=13
+```
+
+
+## Finetuning Vicuna with SkyPilot
+Currently, training requires GPUs with 80GB memory.  See `sky show-gpus --all` for supported GPUs.
+
+We can start the finetuning of Vicuna model on the dummy data [dummy.json](examples/llm-vicuna/dummy-data) **with a single command**. It will automatically find the available cheapest VM on any cloud.
+
+To finetune on your own data, replace the file with your own, or change the line `/data/mydata.json: ./dummy.json` to the path of your own data.
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply examples/llm-vicuna/README.md]: Done. Thanks!

File: examples/llm-vicuna/README.md (hunk 1 above)
Comment (Michaelvll) [reply examples/llm-vicuna/README.md]: Ahh good catch! Fixed. Also, turned wandb off by default.

File: examples/llm-vicuna/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+# Train a 13B model instead of the default 7B
+sky spot launch -n vicuna-7b scripts/train-vicuna.yaml --env WANDB_API_KEY --env MODEL_SIZE=13
+
+# Use *unmanaged* spot instances (i.e., preemptions won't get auto-recovered).
+# Unmanaged spot saves the cost of a small controller VM.  We recommend using managed spot as above.
+sky launch -n vicuna scripts/train-vicuna.yaml --env WANDB_API_KEY
+```
+Currently, such `A100-80GB:8` spot instances are only available on AWS and GCP.
+
+[Optional] **To use on-demand `A100-80GB:8` instances**, which are currently available on Lambda Cloud, Azure, and GCP:
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply examples/llm-vicuna/README.md:71]: Yes. According to the [product page](https://aws.amazon.com/ec2/instance-types/p4/) of `p4de.24xlarge`, it has the offering of spot instance:
> Customers can purchase P4d and P4de instances as On-Demand Instances, Reserved Instances, Spot Instances, Dedicated Hosts or as part of Savings Plan.

**Review by concretevitamin — COMMENTED** [review]
Minor nits. Main comment is to rephrase "finetuning".

File: examples/llm-vicuna/README.md [hunk 4]
```diff
@@ -14,8 +14,8 @@ Install the latest SkyPilot and check your setup of the cloud credentials:
 pip install git+https://github.com/skypilot-org/skypilot.git
 sky check
 ```
-Find the Vicuna finetuning [SkyPilot YAMLs](examples/llm-vicuna/finetune.yaml)
-Find the Vicuna serving [SkyPilot YAMLs](examples/llm-vicuna/serve.yaml)
+Find the Vicuna finetuning [SkyPilot YAMLs](finetune.yaml)
+Find the Vicuna serving [SkyPilot YAMLs](serve.yaml)
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply examples/llm-vicuna/README.md]: Done. Thanks!

File: README.md [hunk 5]
```diff
@@ -26,11 +26,10 @@
 </h3>
 
 ----
-:fire: :dromedary_camel: *News (March, 2023)* :dromedary_camel: :fire: 
-
-- **You can now train and serve the Vicuna model with a single** [**SkyPilot command**](./examples/llm-vicuna/)!
-- **[Vicuna LLM chatbot](https://vicuna.lmsys.org/) trained** [**using SkyPilot**](./examples/llm-vicuna/) **for $300 on spot instances!** 
-- *Serve* your own LLaMA LLM chatbot (not finetuned) on any cloud: [**example**](./examples/llama-llm-chatbots/), [**repo**](https://github.com/skypilot-org/sky-llama)
+:fire: :dromedary_camel: *News* :dromedary_camel: :fire: 
+- [April, 2023] **You can now finetune and serve the Vicuna model with a single** [**SkyPilot command**](./examples/llm/vicuna/)!
```
Comment (concretevitamin) [inline README.md:30]:
```suggestion
- [April, 2023] **[**SkyPilot YAMLs released**](./examples/llm/vicuna/) for finetuning and serving the Vicuna model with a single command** !
```
Not a strong preference, up to you!

File: README.md [hunk 6]
```diff
@@ -26,11 +26,10 @@
...
+- [March, 2023] **[Vicuna LLM chatbot](https://vicuna.lmsys.org/) trained** [**using SkyPilot**](./examples/llm/vicuna/) **for $300 on spot instances!** 
+- [March, 2023] *Serve* your own LLaMA LLM chatbot (not finetuned) on any cloud: [**example**](./examples/llama-llm-chatbots/), [**repo**](https://github.com/skypilot-org/sky-llama)
```
Comment (concretevitamin) [inline README.md:32]: path needs change

File: examples/llm/vicuna/README.md [hunk 7]
```diff
@@ -14,16 +14,15 @@ Install the latest SkyPilot and check your setup of the cloud credentials:
 pip install git+https://github.com/skypilot-org/skypilot.git
 sky check
 ```
-Find the Vicuna finetuning [SkyPilot YAMLs](finetune.yaml)
-Find the Vicuna serving [SkyPilot YAMLs](serve.yaml)
+Find the Vicuna [training SkyPilot YAMLs](train.yaml) and [serving SkyPilot YAMLs](serve.yaml).
```
Comment (concretevitamin) [inline examples/llm/vicuna/README.md:17]:
```suggestion
See the Vicuna SkyPilot YAMLs: for [training](train.yaml) and for [serving](serve.yaml).
```

File: examples/llm/vicuna/README.md [hunk 8]
```diff
@@ -14,16 +14,15 @@ Install the latest SkyPilot and check your setup of the cloud credentials:
...
 
 ## Serve the official Vicuna model by yourself with SkyPilot
 
 1. Start the serving the Vicuna-7B model on a single A100 GPU:
 ```bash
 sky launch -c vicuna-serve -s serve.yaml
 ```
-2. Check the output of the command. There will be a sharable gradio link (like the following). Open it in your browser to chat with Vicuna.
+2. Check the output of the command. There will be a sharable gradio link (like the last line in of the following). Open it in your browser to chat with Vicuna.
```
Comment (concretevitamin) [inline examples/llm/vicuna/README.md:25]:
```suggestion
2. Check the output of the command. There will be a sharable gradio link (like the last line of the following). Open it in your browser to chat with Vicuna.
```

File: examples/llm/vicuna/README.md [hunk 9]
```diff
@@ -41,38 +40,41 @@ sky launch -c vicuna-serve -s serve.yaml --env MODEL_SIZE=13
...
 Currently, training requires GPUs with 80GB memory.  See `sky show-gpus --all` for supported GPUs.
 
-We can start the finetuning of Vicuna model on the dummy data [dummy.json](dummy.json) **with a single command**. It will automatically find the available cheapest VM on any cloud.
+We can start the training of Vicuna model on the dummy data [dummy.json](dummy.json)[^1] **with a single command**. It will automatically find the available cheapest VM on any cloud.
 
-**To finetune on your own data**, replace the file with your own, or change the line `/data/mydata.json: ./dummy.json` to the path of your own data in the SkyPilot YAML .
+**To train on your own data**, replace the file with your own, or change the line `/data/mydata.json: ./dummy.json` to the path of your own data in the [train.yaml](train.yaml).
 
-Steps for finetuning on your cloud(s):
+[^1]: The dummy data was original from the official Vicuna repository, [FastChat](https://github.com/lm-sys/FastChat).
```
Comment (concretevitamin) [inline examples/llm/vicuna/README.md:50]:
```suggestion
[^1]: The dummy data was originally from the official Vicuna repository, [FastChat](https://github.com/lm-sys/FastChat).
```

**Review by concretevitamin — APPROVED** [review]
Kind of feel like instead of `examples/llm/*`, maybe we should put them under `<root>/llm/*` to make them easily identifiable?

**Pull request merged**